### DISCUSSION TITLE
Problems rendering Task after code refactoring.

### DISCUSSION THREAD

**Discussion by b-sia** [charmbracelet/bubbles#275] [Q&A] [answered]
I looked through the codebase for the CLI Kanban tutorial and decided to refactor the source files by moving them to separate directories (typedef.go goes to /typedef, task.go goes to /task, etc).

I moved task.go to the task package. 

```
├── datastore.go
├── form
├── form.go
├── go.mod
├── go.sum
├── main.go
├── model.go
├── README.md
├── storage.json
├── styling.go
├── task
│   └── task.go
└── typedef
    └── typedef.go
```

In model.go, `Task` struct creation in `initList()` are refactored to task.go.

```go
// model.go

func (m *Model) initLists(width, height int) {

    // original task creation code
    for _, value := range columns.Todo {
		task := Task{status: typedef.Todo, title: value.Title, description: value.Description}
		todoItems = append(todoItems, task)
    }

   // task creation post refactoring
   for _, value := range columns.Todo {
		task := task.NewTask(typedef.Todo, value.Title, value.Description)
		todoItems = append(todoItems, task)
   }
}
```
```go
// task/task.go

func NewTask(status typedef.Status, title string, description string) Task {
	return Task{status: status, title: title, description: description}
}
```

After the changes, the Tasks within the columns could not be rendered properly. 

![Screenshot from 2022-10-15 18-52-56](https://user-images.githubusercontent.com/9401015/196217127-5e32afe1-351e-472e-918e-dbeb933f0f9f.png)

How does BubbleTea render the `Task` structs? Does it matter if tasks are created in another package, rather than in the main package?

**Comment by bashbunni** [comment]
Hey @zerovirus123 do you mind pushing the latest changes to your `cli_kanban` repo so I can reproduce the issue? I tried cloning your repo and running it, but it's not giving me the error you're describing. Let me know when this is done and I'll check again!

**Comment by b-sia** [comment]
Hi @bashbunni, I just pushed the latest version to my repo. Sorry for the delay.

**Comment by bashbunni** [comment, marked as answer]
Ah so the issue is just that you renamed the `Title` and `Description` methods which actually need to keep their method signature to implement the `DefaultItem` interface. Check out this portion to see what I'm talking about: https://github.com/charmbracelet/bubbles/blob/cd2593cfb77c4b208aa4ea31dfbe7a070b2e4014/list/defaultitem.go#L64

I've also attached the git diff of my changes to get it running. (Might also include some minor formatting changes from my Go formatter)

[diff.txt](https://github.com/charmbracelet/bubbles/files/9816943/diff.txt)

**Reply by bashbunni** [reply]
Let me know if it's more readable for me to PR the changes - totally up to you. I've got the working version on my machine 😄

**Comment by b-sia** [comment]
Hi, could you please submit a PR? That would be more readable. Thanks.

Brian

**Reply by bashbunni** [reply]
PR'd!
